bin/check-influxdb-metrics.rb

```diff
@@ -79,32 +79,51 @@ class CheckInfluxDbMetrics < Sensu::Plugin::Check::CLI
          long: '--filter=VALUE',
          description: 'Set the name of your filter, for example: `datacenter`'
 
-  def encode_parameters(parameters)
-    encodedparams = Addressable::URI.escape(parameters)
-    "#{config[:db]}&q=" + encodedparams
-  end
+  option :period,
+         long: '--period=VALUE',
+         description: 'Filter by a given day period in minutes',
+         proc: proc { |l| l.to_i },
+         default: 10
 
   def filter_by_environment_when_needed
     config[:tag].nil? && config[:filter].nil? ? '' : " AND \"#{config[:tag]}\" =~ /#{config[:filter]}/"
   end
 
-  def yesterday_query # Reads the value from 10 minutes before yesterday at this time.
-    query = "SELECT sum(\"value\") from \"#{config[:metric]}\" WHERE time > now() - 1455m AND time < now() - 1445m"
+  def base_query
+    "SELECT sum(\"value\") from \"#{config[:metric]}\" "
+  end
+
+  def today_query_for_a_period
+    start_period = '5'; # starts counting 5 minutes before now() to let influxdb time to aggregate the data
```

Comment:
I know much about the compiler of Ruby, but because start_period will be added at the end of the query (which is a string), I thought about setting it as string straight away, to save casting. @andyroyle